From f623c49025a51f38fb2cd506edae7588c8b8073c Mon Sep 17 00:00:00 2001
From: Hugo Osvaldo Barrera <hugo@whynothugo.nl>
Date: Sat, 30 Aug 2025 12:44:41 +0200
Subject: [PATCH] tests: skip speculation enable if force-disabled

Avoids test failure on environments where PR_SPEC_FORCE_DISABLE is
already set by skipping the enable step.

Patch-Source: https://github.com/seveas/python-prctl/pull/40
Fixes: https://github.com/seveas/python-prctl/issues/31
---
 test_prctl.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/test_prctl.py b/test_prctl.py
index df84986..479c085 100644
--- a/test_prctl.py
+++ b/test_prctl.py
@@ -328,6 +328,10 @@ class PrctlTest(unittest.TestCase):
        self.assertTrue(prctl.get_speculation_ctrl(prctl.SPEC_INDIRECT_BRANCH) > 0)
        self.assertRaises(ValueError, prctl.get_speculation_ctrl, 99)
        self.assertRaises(ValueError, prctl.set_speculation_ctrl, 99)
+
+       if prctl.get_speculation_ctrl(prctl.SPEC_STORE_BYPASS) & prctl.SPEC_FORCE_DISABLE:
+            self.skipTest("SPEC_STORE_BYPASS is force-disabled on this environment")
+
        prctl.set_speculation_ctrl(prctl.SPEC_STORE_BYPASS, prctl.SPEC_ENABLE)
        self.assertEqual(prctl.get_speculation_ctrl(prctl.SPEC_STORE_BYPASS) & ~prctl.SPEC_PRCTL, prctl.SPEC_ENABLE)
        prctl.set_speculation_ctrl(prctl.SPEC_STORE_BYPASS, prctl.SPEC_FORCE_DISABLE)
-- 
2.51.0

